Source/Relativity.Test.Helpers/Relativity.Test.Helpers/UserHelpers/CreateUser.cs

```diff
@@ -14,234 +14,276 @@ public static class CreateUser
-			}
-
-			WriteResultSet<User> createResults;
-
 			try
 			{
-				createResults = client.Repositories.User.Create(userDto);
+				const string errorContext = "An error occured when creating a new Relativity User.";
+				int userArtifactId = 0;
+
+				client.APIOptions.WorkspaceID = -1;
+
+				int defaultSelectedFileType = 1;
+				int userTypeCodeTypeId = 3;
+				int documentSkip = 1000003;
+				int skipDefaultPreference = 1000004;
+				int passwordCodeTypeId = 1000005;
+				int sendNewPasswordTo = 1000006;
+
+				int returnPasswordCodeId = FindChoiceArtifactId(client, sendNewPasswordTo, "Return");
+				int passwordCodeId = FindChoiceArtifactId(client, passwordCodeTypeId, "Manually set password");
+				int documentSkipCodeId = FindChoiceArtifactId(client, documentSkip, "Enabled");
+				int documentSkipPreferenceCodeId = FindChoiceArtifactId(client, skipDefaultPreference, "Normal");
+				int defaultFileTypeCodeId = FindChoiceArtifactId(client, defaultSelectedFileType, "Native");
+				int userTypeCodeId = FindChoiceArtifactId(client, userTypeCodeTypeId, userType);
+				int everyoneGroupArtifactId = FindGroupArtifactId(client, "Everyone");
+				int clientArtifactId = FindClientArtifactId(client, clientName);
+
+				User userDto = new User
+				{
+					FirstName = firstName,
+					LastName = lastName,
+					EmailAddress = emailAddress,
+					AdvancedSearchPublicByDefault = true,
+					AuthenticationData = "",
+					BetaUser = false,
+					ChangePassword = true,
+					ChangePasswordNextLogin = false,
+					ChangeSettings = true,
+					Client = new Client(clientArtifactId),
+					DataFocus = 1,
+					DefaultSelectedFileType = new Choice(defaultFileTypeCodeId),
+					DocumentSkip = new Choice(documentSkipCodeId),
+					EnforceViewerCompatibility = true,
+					Groups = new List<kCura.Relativity.Client.DTOs.Group> { new kCura.Relativity.Client.DTOs.Group(everyoneGroupArtifactId) },
+					ItemListPageLength = 25,
+					KeyboardShortcuts = true,
+					MaximumPasswordAge = 0,
+					NativeViewerCacheAhead = true,
+					PasswordAction = new Choice(passwordCodeId),
+					RelativityAccess = relativityAccess,
+					SendPasswordTo = new Choice(returnPasswordCodeId),
+					SkipDefaultPreference = new Choice(documentSkipPreferenceCodeId),
+					TrustedIPs = "",
+					Type = new Choice(userTypeCodeId),
+					Password = password
+				};
+
+				foreach (Int32 groupArtifactId in groupArtifactIds)
+				{
+					userDto.Groups.Add(new kCura.Relativity.Client.DTOs.Group(groupArtifactId));
+				}
+
+				WriteResultSet<User> createResults;
+
+				try
+				{
+					createResults = client.Repositories.User.Create(userDto);
+				}
+				catch (Exception ex)
+				{
+					throw new TestHelpersException(errorContext, ex);
+				}
+
+				// Check for success.
+				if (createResults.Success)
+				{
+					userArtifactId = createResults.Results[0].Artifact.ArtifactID;
+				}
+				else
+				{
+					throw new TestHelpersException("Failed to Create User");

```

Comment:
i think createResults object has more error message which we can throw with the exception which would be helpful in troubleshooting.